### ISSUE TITLE
Running maven tasks does not respect configured automation shell

### ISSUE THREAD

**Issue by PunkIslamist** [microsoft/vscode-maven#563]
**Describe the bug**
Setting `terminal.integrated.automationShell.*` in my workspace, I would expect that running Maven tasks invoke that shell. This is not the case and it uses the configured default shell.

**To Reproduce**
1. Create a VS Code workspace
2. Create a maven project inside that workspace
3. Configure `terminal.integrated.automationShell.*` in your workspace settings with a value different to you default shell
4. Run a Maven task such as `test`
5. The extension invoked the command using the default shell

**Expected behavior**
I expect the extension to invoke the configured automation shell when running tasks, not the default shell.

**Environments (please complete the following information as much as possible):**
 - OS: Windows 10 19042
 - VS Code version: 1.51.1
 - Extension version 0.26.0

**Screenshots**
![image](https://user-images.githubusercontent.com/26367652/101246025-f2f3cb00-3710-11eb-9fe9-38132d2b8ab0.png)

The screenshot shows both settings, workspace and default, and the resulting call in the terminal to the configured default shell.

**Comment by Eskibear** [comment]
`terminal.integrated.automationShell.*` is for tasks and debug. See https://code.visualstudio.com/docs/editor/integrated-terminal#_changing-shell-for-tasks-and-debug Maven extension simply sends commands to default terminals instead of creating and executing **tasks**.

**Comment by PunkIslamist** [comment]
> `terminal.integrated.automationShell.*` is for tasks and debug. See https://code.visualstudio.com/docs/editor/integrated-terminal#_changing-shell-for-tasks-and-debug Maven extension simply sends commands to default terminals instead of creating and executing **tasks**.

Yes, that is exactly the problem. If you take a look at the [original issue that resulted in the setting](https://github.com/microsoft/vscode/issues/78497) it becomes clear that it allows you to use a shell as a user that might not be compatible with commands sent by tools and extensions. Which is precisely why I opened the issue and why the extension, in my opinion, should respect that setting.
In my case I use a shell that throws an error when called by the maven extension so I have to reconfigure my default shell just because of this extension. Everything else is respecting the setting and does not cause problems.
